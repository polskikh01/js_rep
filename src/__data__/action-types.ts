export const PRODUCT = {
    REQUEST: 'PRODUCT_REQUEST',
    SUCCESS: 'PRODUCT_SUCCESS',
    FAILURE: 'PRODUCT_FAILURE',
};

export const BASKET = {
    REQUEST: 'BASKET_REQUEST',
    SUCCESS: 'BASKET_SUCCESS',
    FAILURE: 'BASKET_FAILURE',
};

export const ACCESSORIES = {
    REQUEST: 'ACCESSORIES_REQUEST',
    SUCCESS: 'ACCESSORIES_SUCCESS',
    FAILURE: 'ACCESSORIES_FAILURE',
};

export const COMPLETE = {
    REQUEST: 'COMPLETE_REQUEST',
    SUCCESS: 'COMPLETE_SUCCESS',
    FAILURE: 'COMPLETE_FAILURE',
};