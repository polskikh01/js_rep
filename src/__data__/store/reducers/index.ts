import { combineReducers } from 'redux';

import productReducer, {ProductState} from './product'

export type AppStore = {
    product: ProductState;
};

export default combineReducers<AppStore>({
    product: productReducer
})