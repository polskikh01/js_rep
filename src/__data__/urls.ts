import { getNavigations } from '@ijl/cli';

const navigations = getNavigations('js_rep');

export const baseUrl = navigations['js_rep'];

export const URLs = {
    main: {
        url: navigations['link.js_rep.main'],
    },
    product: {
        url: navigations['link.js_rep.product'],
    },
    accessories: {
        url: navigations['link.js_rep.accessories'],
    },
    complete: {
        url: navigations['link.js_rep.complete'],
    },
    contacts: {
        url: navigations['link.js_rep.contacts'],
    },
    shippingAndPayment: {
        url: navigations['link.js_rep.shippingAndPayment']
    }
}