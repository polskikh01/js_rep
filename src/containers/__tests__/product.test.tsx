import React from 'react';
import {mount} from "enzyme";
import {describe, it, expect} from '@jest/globals'
import {Provider} from 'react-redux'

import {store} from '../../__data__/store'

import Product from '../product/product';

describe('Тестирование всего приложения', () => {
    it('Тестируем рендер Product', () => {
        expect.assertions(1)
        const component = mount(
            <Provider store={store}>
                <Product/>
            </Provider>
        )
        expect(component).toMatchSnapshot()
    });
})