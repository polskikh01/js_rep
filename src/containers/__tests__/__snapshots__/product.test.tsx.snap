// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестирование всего приложения Тестируем рендер Product 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(Product)>
    <Product
      addProduct={[Function]}
      getProducts={[Function]}
      loading={true}
      productItems={Array []}
    >
      <p>
        loading...
      </p>
    </Product>
  </Connect(Product)>
</Provider>
`;
